h2o-core/src/main/java/hex/ModelBuilder.java

```diff
@@ -256,7 +256,12 @@ public void compute2() {
       } finally {
         _parms.read_unlock_frames(_job);
         if (!_parms._is_cv_model) cleanUp(); //cv calls cleanUp on its own terms
-        Scope.exit();
+        if (!_parms._is_cv_model && _parms._fold_column != null) {
```

Comment:
I think this really should not be on ModelBuilder - I will request changes based on these lines.

We can think of better solution together.